### PR TITLE
Add Ceremonies Table

### PR DESCRIPTION
**Pull request by TheShadowEevee**
Add a ceremonies table for https://github.com/purduehackers/passport-issuing-office/pull/52

### CODE REVIEW

**Review by MatthewStanciu — APPROVED** [review]
LGTM assuming build succeeds

**Comment by ImTheSquid** [comment]
I feel like we should have a foreign key or something similar for passport -> ceremony. Opinions?

**Comment by TheShadowEevee** [comment]
With the way it was implemented, that's probably a good idea. I don't actually know how to do that with the rust library however, if you would be able to or point me in the right direction I can do that. The migration was an early piece of the puzzle

File: migration/src/m20240917_210754_ceremonies.rs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#[async_trait::async_trait]
+impl MigrationTrait for Migration {
+    async fn up(&self, manager: &SchemaManager) -> Result<(), DbErr> {
+        manager
+            .create_table(
+                Table::create()
+                    .table(Ceremonies::Table)
+                    .if_not_exists()
+                    .col(
+                        ColumnDef::new(Ceremonies::CeremonyTime)
```
Comment (ImTheSquid) [inline migration/src/m20240917_210754_ceremonies.rs:29]: This feels like this should be a FK from passort

**Review by ImTheSquid — CHANGES_REQUESTED** [review]

**Comment by ImTheSquid** [comment]
> With the way it was implemented, that's probably a good idea. I don't actually know how to do that with the rust library however, if you would be able to or point me in the right direction I can do that. The migration was an early piece of the puzzle

Look at the other migration files, they have examples on how to do it. Also [docs](https://www.sea-ql.org/SeaORM/docs/0.4.x/generate-database-schema/create-table/)

**Comment by MatthewStanciu** [comment]
Here's an example: https://github.com/purduehackers/id/blob/a59b67bcb7893ff8f377af5f37938dfa81c4fb05/migration/src/m20220101_000001_create_table.rs#L98

File: migration/src/m20240917_210754_ceremonies.rs (hunk 1 above)
Comment (TheShadowEevee) [reply migration/src/m20240917_210754_ceremonies.rs:29]: Added

**Review by ImTheSquid — COMMENTED** [review]
Did you update the schema such that the string is converted to a timestamp for the inserted row?

**Comment by TheShadowEevee** [comment]
I did that before the current ceremony_time, so yes it is. Should I change the timestamp to a string? Or do you mean passport needs to be changed to be a timestamp

File: migration/entity/src/entities/passport.rs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+pub enum Relation {
+    #[sea_orm(
+        belongs_to = "Entity",
+        from = "Column::CeremonyTime",
+        to = "Column::CeremonyTime",
+        on_update = "Cascade",
+        on_delete = "Cascade"
+    )]
+    SelfRef,
```
Comment (ImTheSquid) [inline migration/entity/src/entities/passport.rs]: This is incorrect, it's self referential. Your `to` is wrong.

**Review by ImTheSquid — CHANGES_REQUESTED** [review]

File: migration/entity/src/entities/passport.rs (hunk 2 above)
Comment (TheShadowEevee) [reply migration/entity/src/entities/passport.rs]: Whoops... Fixed

**Comment by ImTheSquid** [comment]
Did those changes migrate successfully?

**Comment by TheShadowEevee** [comment]
Yes

File: migration/entity/src/entities/ceremonies.rs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+pub struct Model {
+    #[sea_orm(primary_key, auto_increment = false)]
+    pub ceremony_time: DateTime,
+    pub total_slots: i32,
+    pub open_registration: bool,
+}
+
+#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
+pub enum Relation {
+    #[sea_orm(has_one = "super::passport::Entity")]
```
Comment (ImTheSquid) [inline migration/entity/src/entities/ceremonies.rs]: This doesn't seem right, a ceremony has many passports, not one

Comment (TheShadowEevee) [reply migration/entity/src/entities/ceremonies.rs]: I did something dumb, fixed

**Review by ImTheSquid — APPROVED** [review]
Everything looks good!

**Pull request merged**